crates/nu-cli/src/git.rs

```diff
@@ -2,13 +2,8 @@ use crate::prelude::*;
 
 pub fn current_branch() -> Option<String> {
     if let Ok(config) = crate::data::config::config(Tag::unknown()) {
-        let use_starship = match config.get("use_starship") {
-            Some(b) => match b.as_bool() {
-                Ok(b) => b,
-                _ => false,
-            },
-            _ => false,
-        };
+        let use_starship =
+            matches!(config.get("use_starship"), Some(b) if matches!(b.as_bool(), Ok(b) if b));
```

Comment:
I think I've seen this a few times now. Another way to write this would be to not using any `match` at all, but take advantage of some of the lovely functionality that `Option` has:

```rust
let use_starship = config
    .get("use_starship")
    .and_then(Value::as_bool)
    .unwrap_or(false);
```

I find this easier to follow than the nested `matches`. There's just a lot of parentheses and `if`'s that are hard to "match" together.

Reply:
@thegedge - agreed. I think that style here makes it clearer what's happening, though I love that we're cleaning up unnecessary matches, generally.

Reply:
I think this is more readable as well, but it doesn't work:

```rust
error[E0271]: type mismatch resolving `<for<'r> fn(&'r nu_protocol::value::Value) -> std::result::Result<bool, nu_errors::ShellError> {nu_protocol::value::Value::as_bool} as std::ops::FnOnce<(&nu_protocol::value::Value,)>>::Output == std::option::Option<_>`
   --> crates/nu-cli/src/cli.rs:723:10
    |
723 |         .and_then(Value::as_bool)
    |          ^^^^^^^^ expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected enum `std::result::Result<bool, nu_errors::ShellError>`
               found enum `std::option::Option<_>`
```

Reply:
I'll have to look at the documentation and see if there's another function to chain in there that deals with this.

Reply:
I think this would be what we need @thegedge and @jonathandturner:

```rust
let use_starship = config
    .get("use_starship")
    .map(|x| x.is_true())
    .unwrap_or(false);
```

Reply:
Ah yes, can't mix Results and Options. Looks good with a `map` too 🙂